client/components/autocomplete_select/index.coffee

```diff
@@ -25,7 +25,7 @@ module.exports.AutocompleteSelect = AutocompleteSection = React.createClass
     @autocomplete = new Autocomplete _.extend _.pick(@props, 'url', 'filter'),
       el: $(@refs.input?.getDOMNode())
       selected: (e, item) =>
-        @setState value: item.value, id: item.id
+        _.defer => @setState value: item.value, id: item.id
```

Comment:
Would probably be good to leave a comment here—something simple like # Defering because of click race condition to avoid accidentally removing it later.